libraries/AP_Arming/AP_Arming.cpp

```diff
@@ -168,10 +168,27 @@ bool AP_Arming::barometer_checks(bool report)
 {
     if ((checks_to_perform & ARMING_CHECK_ALL) ||
         (checks_to_perform & ARMING_CHECK_BARO)) {
-        if (!AP::baro().all_healthy()) {
+        const AP_Baro *baro = AP_Baro::get_singleton();
+        if (!baro->all_healthy()) {
             check_failed(ARMING_CHECK_BARO, report, "Barometer not healthy");
             return false;
         }
+        if (checks_to_perform & ARMING_CHECK_GPS) {
+            const AP_GPS *gps = AP_GPS::get_singleton();
+            if (gps->status() >= AP_GPS::GPS_OK_FIX_3D) {
+                if (gps->get_vdop() < 170 && gps->num_sats() > 8) {
+                    float baro_altasl = baro->get_altitude_difference(101325.0f, baro->get_pressure());
```

Comment:
this assumes a 'standard' atmospheric pressure of 101.325. The real pressure varies a lot from this so this will generate a lot of false arming failures. The 100m margin is not nearly enough for real world pressure changes.

Reply:
Increase the margin?
If you've a better idea, I can work on it.